qcodes_contrib_drivers/drivers/RohdeSchwarz/private/HMP.py

```diff
@@ -0,0 +1,118 @@
+                           label='Target voltage output',
+                           set_cmd=partial(self.send_cmd, "set_voltage"),
+                           get_cmd=partial(self.send_cmd, "set_voltage", None),
+                           get_parser=float,
+                           unit='V',
+                           vals=vals.Numbers(0, 32.050)
+                           )
+        self.add_parameter("set_current",
+                           label='Target current output',
+                           set_cmd=partial(self.send_cmd, "set_current"),
+                           get_cmd=partial(self.send_cmd, "set_current", None),
+                           get_parser=float,
+                           unit='A',
+                           vals=vals.Numbers(0.5e-3, self.max_current)
+                           )
+        self.add_parameter('state',
+                           label='Output enabled',
+                           set_cmd=partial(self.send_cmd, "state"),
+                           get_cmd=partial(self.send_cmd, "state", None),
+                           val_mapping={'ON': 1, 'OFF': 0},
+                           vals=vals.Enum('ON', 'OFF')
+                           )
+        self.add_parameter("voltage",
+                           label='Measured voltage',
+                           get_cmd=partial(self.send_cmd, "voltage", None),
+                           get_parser=float,
+                           unit='V',
+                          )
+        self.add_parameter("current",
+                           label='Measured current',
+                           get_cmd=partial(self.send_cmd, "current", None),
+                           get_parser=float,
+                           unit='A',
+                           )
+        self.add_parameter("power",
+                           label='Measured power',
+                           get_cmd=self.get_power,
+                           get_parser=float,
+                           unit='W',
+                           )
+
+    def get_max_current(self):
+        if self.parent.model_no > 4000:
+            return 10
+        elif self.parent.model_no == 2020 and self.channel == 1:
+            return 10
+        return 5
+
+    def send_cmd(self, param, value):
+        self.write(f"INSTrument:NSELect {self.channel:d}")
+        if value is None:
+            return self.ask(f"{self.get_scpi_command(param)}?")
+        else:
+            return self.write(f"{self.get_scpi_command(param)} {value}")
+
+
+    @staticmethod
+    def get_scpi_command(param):
+        commands = {
+            "set_voltage": "SOURce:VOLTage:LEVel:IMMediate:AMPLitude",
+            "set_current": "SOURce:CURRent:LEVel:IMMediate:AMPLitude",
+            "state": "OUTPut:STATe",
+            "voltage": "MEASure:SCALar:VOLTage:DC",
+            "current": "MEASure:SCALar:CURRent:DC",
+        }
+        return commands.get(param)
+
+    def get_power(self):
```

Comment:
I don't think the user should ever need to call this directly so I would make it protected with a name like `_get_power`